app/controllers/orders_controller/order_status.rb

```diff
@@ -0,0 +1,26 @@
+class OrdersController
+  class OrderStatus
+    def initialize(current_user)
+      @current_user = current_user
+    end
+
+    def save
+      @current_user.orders.each do |order|
+        next if order.status == "Delivered"
+        status = get_status(order.created_at) || order.status
```

Comment:
Why are  we assigning `order.status` to `status` here and then updating order attribute status with status again.